tools/update_certs.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python3
+# Copyright 2024 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tool for updating Cobalt SSL Certificates.
+
+Usage:
+  update_certs.py -r <cobalt_repo> -b <openssl_bin> -p <pem_files>
+
+Args:
+    cobalt_repo: The path to the Cobalt repository root.
+    openssl_bin: The path to the OpenSSL binary, defaults to "/usr/bin/openssl"
+      when not specified.
+    pem_files: The paths to the source pem files.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+
+_CERTS_PATH = 'cobalt/content/ssl/certs'
+_BUILD_PATH = 'cobalt/network/certs.gni'
+
+
+def SplitCertSet(concatenated_cert_file):
+  """Split concatenated .pem file into a list of individual certs."""
+  begin = '-----BEGIN CERTIFICATE-----'
+  end = '-----END CERTIFICATE-----'
+
+  cert_set = []
+  current_cert = ''
+  in_cert = False
+
+  for line in concatenated_cert_file:
+    if not in_cert and line.find(begin) != -1:
+      in_cert = True
+      current_cert += line
+    elif in_cert and line.find(end) != -1:
+      in_cert = False
+      current_cert += line
+      cert_set.append(current_cert)
+      current_cert = ''
+    else:
+      if in_cert:
+        current_cert += line
+
+  return cert_set
+
+
+def ComputeCertFilename(openssl_bin, cert):
+  """Use x509 to get the hashed filename for the individual cert."""
+  command_args = [openssl_bin, 'x509', '-subject_hash', '-noout']
+
+  with subprocess.Popen(
+      command_args, stdout=subprocess.PIPE, stdin=subprocess.PIPE,
+      shell=False) as process:
+    output, err = process.communicate(cert.encode('utf-8'))
+
+    if process.returncode:
+      cmd_args = ' '.join(command_args)
+      print(
+          f'OpenSSL failed [{process.returncode}]: {cmd_args}\n{output}\n{err}')
+
+    return output.decode().rstrip() + '.0'
+
+  return ''
+
+
+def UpdateBuild(build_file, cert_filepaths):
+  cert_filepaths = list(cert_filepaths)
+  cert_filepaths.sort()
+
+  # Removes existing build file
+  if os.path.exists(build_file):
+    os.remove(build_file)
+
+  # Adds build file
+  certs = 'certs = [\n  "//'
```

Comment:
I think it would be more readable if you did this line by line—I at least fine the newlines in the middle of the strings hard to parse

Reply:
that's what i initially had but our py lint presubmits HATES the fact that I was concatenating lines rather than using join.

Reply:
Something like the below wouldn't work?
```
cert_lines = ['certs = [\n']
cert_lines += [f'  "//{path}",\n' for path in cert_filepaths]
cert_lines += [']\n']
with open(...) as f:
  f.writelines(cert_lines)
```

Reply:
that works, thanks, it avoids the pre-submit complaints by using a list.